.github/workflows/build_and_push.yml

```diff
@@ -0,0 +1,208 @@
+      - uses: rrbutani/use-nix-shell-action@v1.1.0
+        with:
+            file: shell.nix
+
+      - name: Check if the chart follows a valid semver version
+        run: |
+          BRANCH=${{ github.ref_name }}
+          ./scripts/validate-chart-version.sh --branch $BRANCH
+
+      - name: Run chart-testing lint
+        run: |
+          ct lint --config ct.yaml
+
+  unit-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Unit test
+        run: make test
+
+      - name: Upload Coverage Report
+        uses: codecov/codecov-action@v4  
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          file: ./coverage.txt
+          name: coverage-$(date +%s)
+          flags: unittests
+
+  bdd-tests:
+    runs-on: ubuntu-latest
+    needs: ['unit-tests']
+    strategy:
+      fail-fast: true
+      matrix:
+        kubernetes: [v1.27.3]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Go 1.19
+        uses: actions/setup-go@v4
+        with:
+          go-version: 1.19.9
+          cache: false
+
+      - name: Build images locally
+        run: make zfs-driver-image || exit 1;
+
+      - name: Setup zfs pools
+        run: |
+          sudo apt-get install zfsutils-linux -y
+          truncate -s 100G /tmp/disk.img
+          sudo zpool create zfspv-pool $(sudo losetup -f /tmp/disk.img --show)
+
+      - name: Setup Minikube-Kubernetes
+        uses: medyagh/setup-minikube@latest
+        with:
+          cache: false
+          minikube-version: 1.31.1
+          driver: none
+          kubernetes-version: ${{ matrix.kubernetes }}
+          cni: calico
+          start-args: '--install-addons=false'
+
+      - name: Setting environment variables
+        run: |
+          echo "KUBECONFIG=$HOME/.kube/config" >> $GITHUB_ENV
+          echo "OPENEBS_NAMESPACE=openebs" >> $GITHUB_ENV
+
+      - name: bootstrap
+        run: make bootstrap
+
+      - name: Running tests
+        run: |
+          make ci
+          make sanity 
+
+      - name: Upload CI Test Coverage Report
+        uses: codecov/codecov-action@v4  
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
+          file: ./tests/bdd_coverage.txt
+          name: coverage-bdd_coverage-$(date +%s)
+          flags: bddtests
+
+  csi-driver:
+    runs-on: ubuntu-latest
+    needs: ['lint', 'bdd-tests']
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set Image Org
+        run: |
+            [ -z "${{ secrets.IMAGE_ORG }}" ] && IMAGE_ORG=openebs || IMAGE_ORG=${{ secrets.IMAGE_ORG}}
+            echo "IMAGE_ORG=${IMAGE_ORG}" >> $GITHUB_ENV
+
+      - name: Set IMAGE_TAG and BRANCH
+        run: |
+          BRANCH=${{ github.ref_name }}
+          echo "BRANCH=$BRANCH" >> $GITHUB_ENV
+          echo "IMAGE_TAG=$(awk -F': ' '/^version:/ {print $$2}' deploy/helm/charts/Chart.yaml)" >> $GITHUB_ENV
```

Comment:
There inconsistency in the indentation, down below and above as well. It should be 2 spaces, because it's 2 spaces on the rest of the workflow files.

Reply:
Right. I used a formatter to indent it now.

Reply:
LGTM, thanks.